saleor/checkout/complete_checkout.py

```diff
@@ -1339,63 +1345,29 @@ def complete_checkout(
             order = Order.objects.get_by_checkout_token(checkout_info.checkout.token)
             return order, False, {}
 
-        # Run pre-payment checks to make sure, that nothing has changed to the
-        # checkout, during processing payment.
+        # We need to refetch the checkout info to ensure that we process checkout
+        # for correct data.
+        lines, _ = fetch_checkout_lines(checkout, skip_recalculation=True)
+        checkout_info = fetch_checkout_info(checkout, lines, discounts, manager)
+
         checkout_info.checkout.voucher_code = None
-        _, _, post_payment_order_data = complete_checkout_pre_payment_part(
```

Comment:
Are you sure of this? We have two separate blocks of locks. It means that between these blocks something can happen to the checkout, like adding more lines to the checkout etc. Pre_payment_part has checks like validate if checkout is fully_paid, etc.

Reply:
We were discussing this with @patrys and we agreed that the second part should only fetch the checkout again without doing any calculations or validations. Any mismatch should be done by the shop owner. @patrys could you confirm? Or maybe I misunderstood something?

Reply:
@korycins At this stage we've already collected the payment, we can either continue with the current state of checkout, or attempt a refund (which can also fail). I think a good first approach is to create the order and let CX deal with any discrepancies as those will be edge cases and any outstanding unpaid balance will be visible.